docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -457,30 +471,53 @@ Nonetheless, the providers effectively delegate most of the token-associated sta
 ====
 
 [[table]]
-.Token authentication mechanism comparison
+.Comparison of features by token authentication mechanism or extension
 |===
-^|Feature required 3+^| Authentication mechanism
+ifndef::quarkus-elytron-security-oauth2[ ^|Feature required 3+^| Authentication mechanism]
+ifdef::quarkus-elytron-security-oauth2[ ^|Feature required 2+^| Authentication mechanism]
 
-^| ^s|`quarkus-oidc` ^s|`quarkus-smallrye-jwt` ^s| `quarkus-elytron-security-oauth2`
+^| ^s|`quarkus-oidc` ^s|`quarkus-smallrye-jwt`
+ifndef::quarkus-elytron-security-oauth2[ ^s|`quarkus-elytron-security-oauth2`]
 
-s|Bearer JWT verification ^|Local verification or introspection  ^|Local verification  ^|Introspection
+s|Bearer JWT verification ^|Local verification or introspection  ^|Local verification
+ifndef::quarkus-elytron-security-oauth2[ ^|Introspection]
 
-s|Bearer opaque token verification ^|Introspection   ^|No  ^|Introspection
-s|Refreshing `JsonWebKey` set to verify JWT tokens  ^|Yes   ^|Yes  ^|No
-s|Represent token as `Principal`  ^|Yes   ^|Yes  ^|Yes
-s|Inject JWT as MP JWT  ^|Yes   ^|Yes  ^|No
+s|Bearer opaque token verification ^|Introspection   ^|No
+ifndef::quarkus-elytron-security-oauth2[ ^|Introspection]
 
-s|Authorization code flow  ^| Yes  ^|No  ^|No
-s|Multi-tenancy ^| Yes  ^|No  ^|No
-s|User information support  ^| Yes  ^|No  ^|No
-s|PEM key format support  ^|No   ^|Yes  ^|No
+s|Refreshing `JsonWebKey` set to verify JWT tokens  ^|Yes   ^|Yes
+ifndef::quarkus-elytron-security-oauth2[ ^|No]
 
-s|SecretKey support ^|No ^|In JSON Web Key (JWK) format ^|No
-s|Inner-signed and encrypted or encrypted tokens ^|Introspection ^|Local verification ^|Introspection
-s|Custom token verification ^|No ^|With injected JWT parser ^|No
-s|JWT as a cookie support ^|No ^|Yes ^|Yes
-|===
+s|Represent token as `Principal`  ^|Yes   ^|Yes
+ifndef::quarkus-elytron-security-oauth2[ ^|Yes]
+
+s|Inject JWT as MP JWT  ^|Yes   ^|Yes
```

Comment:
this is extremely hard to read/maintain. Can we dont find a better way or handle this seperately?

Reply:
I agree, @maxandersen. Unfortunately, asciidoc doesn't offer a way to simply hide a column. 
I can make two tables, one for community and another for product.
WDYT?

Reply:
@rolfedh The table https://quarkus.io/guides/security-authentication-mechanisms#oidc-jwt-oauth2-comparison looks well structured, how will it look now ? May be we can keep it as is upstream and you can tweak it at the product level ?

Reply:
If this updated version isn't acceptable, I can move the table to a separate file, include the file, create a product variant of the file, and use conditionals to switch between the two files. Community users would see only the file that contains the  community version of the table. The product version of the table would be maintained entirely in the product repo. This adds a little bit of maintenance overhead requiring reliable reviews and updates each release.

Reply:
> looks well structured, how will it look now ?

For users, the appearance remains unchanged. 
For contributors working on the asciidoc source, I have made improvements based on feedback I received here.
The current version, with each cell on a new line, follows [the pattern prescribed by the Asciidoctor Docs](https://docs.asciidoctor.org/asciidoc/latest/tables/build-a-basic-table/#create-a-table-with-two-columns-and-three-rows) and is more intuitive for contributors than the original.

Reply:
However, if this improved version still isn't acceptable, we can do this instead:

I could move the table to a separate file in community, include the file, create a product-only variant of the file, and use conditionals to switch between the two files. 

Community users would see only the community version of the file. 
In the product docs we would have a checklist task to compare the two versions of the files and manually update the product file with any significant upstream changes.

Reply:
@rolfedh Makes sense, thanks

Reply:
Add a note in the comment.